### PR TITLE
Adding an endpoint for currency

### PR DESCRIPTION
**Pull request by AymericNoel**
I added a last endpoint: CurrencyEndpoint because this endpoint allows to get the decimals of each token supported by CoinbaseCustody and thus allow the conversion of the balance and the amount of transaction in decimal and not in native token units.

### CODE REVIEW

File: src/Trakx.Coinbase.Custody.Client/Endpoints/CurrencyEndpoint.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            _client = client;
+        }
+
+        /// <inheritdoc />
+        public async Task<PagedResponse<Currency>> ListCurrenciesAsync(string? currency = null, CancellationToken cancellationToken = default)
+        {
+            return await _client
+                .Request("currencies")
+                .SetQueryParam("symbol",currency)
```
Comment (monsieurleberre) [inline src/Trakx.Coinbase.Custody.Client/Endpoints/CurrencyEndpoint.cs:23]: I think it is `.AppendPathSegment()` instead
I tried and query params don't seem supported.

File: src/Trakx.Coinbase.Custody.Client.Tests/Unit/Endpoints/CurrencyEndpointTest.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                .WithVerb(HttpMethod.Get);
+        }
+
+        [Fact]
+        public async Task ListCurrenciesAsync_should_deserialize_response_to_model()
+        {
+            var sampleResponse = await SampleResponseHelper.GetSampleResponseContent("PagedResponseCurrency");
+            HttpTest.RespondWith(sampleResponse);
+
+            var response = await _currencyEndpoint.ListCurrenciesAsync("btc");
```
Comment (monsieurleberre) [inline src/Trakx.Coinbase.Custody.Client.Tests/Unit/Endpoints/CurrencyEndpointTest.cs:42]: So maybe not the best example as you pass in a symbol and return all.
Not knowing what it should do I checked manually. It looks like it should only return one currency.

File: src/Trakx.Coinbase.Custody.Client/Endpoints/CurrencyEndpoint.cs (hunk 1 above)
Comment (AymericNoel) [reply src/Trakx.Coinbase.Custody.Client/Endpoints/CurrencyEndpoint.cs:23]: image

When I debug, i have the correct query parameters... are you sure ?

Comment (AymericNoel) [reply src/Trakx.Coinbase.Custody.Client/Endpoints/CurrencyEndpoint.cs:23]: and the tests about query parameters have succeeded

Comment (monsieurleberre) [reply src/Trakx.Coinbase.Custody.Client/Endpoints/CurrencyEndpoint.cs:23]: image
VS
image

I see that your code is well tested, but it is testing its own (and that is perfectly OK for a unit test). However I think it doesn't seem to match with what the outside world expects.

**Pull request merged**